packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,282 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+
+// Standard header margin, determined from SwiftUI's Forms in iOS 14.2 SDK.
+const EdgeInsetsDirectional _kDefaultHeaderMargin =
+    EdgeInsetsDirectional.fromSTEB(16.5, 16.0, 16.5, 10.0);
+
+// Used for iOS "Inset Grouped" margin, determined from SwiftUI's Forms in
+// iOS 14.2 SDK.
+const EdgeInsetsDirectional _kDefaultInsetGroupedRowsMargin =
+    EdgeInsetsDirectional.fromSTEB(16.5, 0.0, 16.5, 16.5);
+
+// Used for iOS "Inset Grouped" border radius, estimated from SwiftUI's Forms in
+// iOS 14.2 SDK.
+// TODO(edrisian): This should be a rounded rectangle once that shape is added.
+const BorderRadius _kDefaultInsetGroupedBorderRadius =
+    BorderRadius.all(Radius.circular(10.0));
+
+// Used to differentiate the edge-to-edge section with the centered section.
+enum _CupertinoFormSectionType { base, insetGrouped }
+
+/// An iOS-style form section.
+///
+/// The base constructor for [CupertinoFormSection] constructs an
+/// edge-to-edge style section which includes an iOS-style header, footer, rows,
+/// the dividers between rows, and borders on top and bottom of the rows.
+///
+/// The [CupertinoFormSection.insetGrouped] constructor creates a round-edged and
+/// padded section that is commonly seen in notched-displays like iPhone X and
+/// beyond. Creates an iOS-style header, footer, rows, and the dividers
+/// between rows. Does not create borders on top and bottom of the rows.
+///
+/// The [header] parameter sets the form section header. The section header lies
+/// above the [children] rows, with margins that match the iOS style.
+///
+/// The [children] parameter is required and sets the list of rows shown in
+/// the section. The [children] parameter takes a list, as opposed to a more
+/// efficient builder function that lazy builds, because forms are intended to
+/// be short in row count. It is recommended that only [CupertinoFormRow] and
+/// [CupertinoTextFormFieldRow] widgets be included in the [children] list in
+/// order to retain the iOS look.
+///
+/// The [margin] parameter sets the spacing around the content area of the
+/// section encapsulating [children].
+///
+/// The [sectionBodyDecoration] parameter sets the decoration around [children].
```

Comment:
This is a reasonable name. Though we've been so overwhelmingly using the name `decoration` in other widgets it might be beneficial to just use that simple name for consistency.